### PR TITLE
fix limit type

### PR DESCRIPTION
**Pull request by bendavies**
fix limit being parsed to resolveMaxPerPage - must be ?int

| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | #8439 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php [hunk 1]
```diff
@@ -61,9 +61,12 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
 
         if ($paginator instanceof Pagerfanta) {
             $request = $requestConfiguration->getRequest();
+            if (null !== $limit = $request->query->get('limit')) {
+                $limit = (int)$limit;
+            }
 
             $paginator->setMaxPerPage($this->resolveMaxPerPage(
-                $request->query->get('limit'),
+                $limit,
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php]: Won't the `$request->query->getInt('limit', null)` work?

Comment (bendavies) [reply src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php]: hah, i didn't see that method. will update

Comment (bendavies) [reply src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php]: @lchrusciel done

Comment (lchrusciel) [reply src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php]: Didn't know that it will break so many things... sorry :(

Comment (bendavies) [reply src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php]: I'll take a look

**Comment by bendavies** [comment]
sorry @lchrusciel i've not had time to look into this yet. I've not got behat tests running locally.
Are there docs explaining how to?

**Comment by lchrusciel** [comment]
And we could stay with your previous proposal, sorry about that. Maybe we will manage to add to 1.0

**Comment by bendavies** [comment]
why? yours seems better. I've not investigated why the tests are failing yet

**Comment by pamil** [comment]
@bendavies `getInt` is implemented as `return (int) $this->get($key, $default)` while we need something like `return $this->has($key) ? (int) $this->get($key) : null;` because `(int) null === 0`.

**Comment by bendavies** [comment]
@pamil ah, are you sending `limit=` in a url somewhere?

**Comment by bendavies** [comment]
I've switched to my original implementation.

**Comment by bendavies** [comment]
@lchrusciel @pamil all good

File: src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php [hunk 2]
```diff
@@ -61,9 +61,12 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
 
         if ($paginator instanceof Pagerfanta) {
             $request = $requestConfiguration->getRequest();
+            if (null !== $limit = $request->query->get('limit')) {
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php]: How about making it a one-liner `$limit = $request->query->has('limit') ? $request->query->getInt('limit') : null`? A little bit more readable for me (no assignment in a condition).

Comment (bendavies) [reply src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php]: done

**Comment by lchrusciel** [comment]
Thanks Ben!

**Pull request merged**